### DISCUSSION TITLE
KNeighborsTimeSeriesRegressor?

### DISCUSSION THREAD

**Discussion by sly8787** [alan-turing-institute/sktime#2645] [Ideas]
Hi guys, are there any plans to implement an KNeighborsTimeSeriesRegressor-model perhaps?

Best regards

**Comment by fkiraly** [comment]
@TonyBagnall, @chrisholder, @MatthewMiddlehurst?

Btw, I think there's already a way to get what you want. It should work if you use any of the metrics in the `distances` module as the value of `metric` in `sklearn`'s `KNeighborsRegressor`. You need to pass a 3D numpy array, I think.

Alternatively, you can select `metric="precomputed"` and pass the output of `transform` to `sklearn`'s `KNeighborsRegressor`.

**Comment by fkiraly** [comment]
Also, I just looked into it - with the current modular base class interface, it took me 10min to make a first refactor draft:
https://github.com/alan-turing-institute/sktime/pull/2647

Testing and feedback appreciated.

**Comment by fkiraly** [comment]
now merged, will be in 0.12.0

**Reply by sly8787** [reply]
Damn good job! 
Thank you very much. Hopefully in the near future I can make my contribution to this great project.
